security/monitoring.adoc

```diff
@@ -0,0 +1,204 @@
+
+4
+----
+
+[[NOTE]]
+====
+When a namespace is deleted, its events are deleted along with it. Events also expire and are removed to prevent filling up *etcd*. Events are not stored
+as a permanent record and frequent polling is necessary to capture statistics over time.
+====
+
+[[security-monitoring-logs]]
+== Cluster Logs
+
+[[security-monitoring-service-logs]]
+=== Service Logs
+
+{product-title} produces logs with each *systemd* service that is running on a host:
+
+ifdef::openshift-origin[]
+- origin-master-api
+- origin-master-controllers
+- etcd
+- origin-node
+endif::[]
+ifdef::openshift-enterprise[]
+- atomic-openshift-master-api
+- atomic-openshift-master-controllers
+- etcd
+- atomic-openshift-node
+endif::[]
+
+These logs provide a relatively verbose record of component operations
+directed more at debugging purposes than security auditing. They can be
+retrieved per host with `*journalctl*` or, in clusters with the aggregated
+logging stack deployed, in the logging _.operations_ indexes (possibly
+in the xref:../install_config/aggregate_logging.adoc#aggregated-ops[Ops
+cluster]) as a cluster administrator.
+
+[[security-monitoring-audit-log]]
+=== Master API Audit Log
+
+For security auditing purposes, there is an option to enable
+audit logging of the sequence of master API requests by individual
+users, administrators, or system components. The audit log can be
+xref:../install_config/master_node_configuration.adoc#master-node-config-audit-config[configured
+for the master API] to log separately to a file on each master host or
+(if no file is configured) to be included in the service's journal
+(where entries can easily be identified by searching for *"AUDIT"*).
+
+xref:../install_config/master_node_configuration.adoc#master-node-config-audit-config[Audit
+log entries] consist of one line recording each REST request when it is
+received and one line with the HTTP response code when it completes. For
+example, here is a record of the system administrator requesting a list
+of nodes:
+
+[source, bash]
+----
+2017-10-17T13:12:17.635085787Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" ip="192.168.122.156" method="GET" user="system:admin" groups="\"system:cluster-admins\",\"system:authenticated\"" as="<self>" asgroups="<lookup>" namespace="<none>" uri="/api/v1/nodes"
+2017-10-17T13:12:17.636081056Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" response="200"
+----
+
+For this example, it may be useful to poll the log periodically by using the following
+command to see if the number of failed requests:
+
+[source, bash]
+----
+$ tail -5000 /var/log/openshift-audit.log \
+  | grep -Po 'response="..."' \
+  | sort | uniq -c | sort -rn
+
+   3288 response="200"
+      8 response="404"
+      6 response="201"
+----
+
+Most valid requests succeed with a 200 or 201 response code. 401 and
+403 are simple redirects, and 404 is typically a benign request for a
+resource that does not exist, but return code 400 may be of
```

Comment:
"but return code 400 may be of interest as it indicates a malformed request, which should not occur with most clients."

IBM Styleguide recommends not using 'may' instead use 'might'.

Also, this should probably be a new sentence instead of a 'but'.